### ISSUE TITLE
Model overfitting when finetuned on smaller data

### ISSUE THREAD

**Issue by diksham-flwls** [joonson/syncnet_trainer#20]
Hi
I am trying to finetune the model on a smaller dataset of 450 samples of greyhead renders that look like this
<img width="711" alt="Screenshot 2024-02-06 at 10 42 08 AM" src="https://github.com/joonson/syncnet_trainer/assets/142529543/cb637136-eaf9-410a-bb3c-d50973a064e1">

My training loss seems to be converging but the validation loss is diverging.
I am freezing everything but the final layers but otherwise using the exact same code as the repo.
<img width="349" alt="Screenshot 2024-02-07 at 7 41 56 PM" src="https://github.com/joonson/syncnet_trainer/assets/142529543/863e4fbe-3c08-437e-9817-239ea8540a1c">
<img width="350" alt="Screenshot 2024-02-07 at 7 42 06 PM" src="https://github.com/joonson/syncnet_trainer/assets/142529543/b80e6b39-740d-4375-8524-ae596886ac28">
When I plot the euclidean distance of the FC layer outputs I find that the distribution is pretty overlapping for matched (green) and unmatched pairs (red).
![matched_and_unmatched_pair_dist](https://github.com/joonson/syncnet_trainer/assets/142529543/d9ea466b-d6d3-454a-b330-065c33b79b9d)

**Comment by silence-will20** [comment]
> 你好， 我正在尝试在一个较小的数据集上对模型进行微调，该数据集包含 450 个灰头渲染样本，如下所示 <img alt="截图于 2024-02-06 上午 10:42:08" width="711" src="https://private-user-images.githubusercontent.com/142529543/303446931-cb637136-eaf9-410a-bb3c-d50973a064e1.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTcwMzkxMjksIm5iZiI6MTc1NzAzODgyOSwicGF0aCI6Ii8xNDI1Mjk1NDMvMzAzNDQ2OTMxLWNiNjM3MTM2LWVhZjktNDEwYS1iYjNjLWQ1MDk3M2EwNjRlMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkwNVQwMjIwMjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00MGIzNzA2NGY5MWJhZWNiZjdiMTg4OTJjNzdkNTg4YjAxMTVhNTA2N2Y4MGY0NTFhNjE2NTkxYmY2NmQ4YTUyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Hyybjf8L7bxTZFV_zhK6MiNS2RhJKI5Xb_2wGTfiQ-Q">
> 
> 我的训练损失似乎正在收敛，但验证损失却在发散。 我冻结了除最后几层之外的所有内容，但其他部分使用的代码与代码库完全相同。 当我绘制全连接层输出的欧氏距离时，我发现匹配对（绿色）和不匹配对（红色）的分布相当重叠。 <img alt="截图于 2024-02-07 下午 7 点 41 分 56 秒" width="349" src="https://private-user-images.githubusercontent.com/142529543/303447434-863e4fbe-3c08-437e-9817-239ea8540a1c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTcwMzkxMjksIm5iZiI6MTc1NzAzODgyOSwicGF0aCI6Ii8xNDI1Mjk1NDMvMzAzNDQ3NDM0LTg2M2U0ZmJlLTNjMDgtNDM3ZS05ODE3LTIzOWVhODU0MGExYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkwNVQwMjIwMjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01MzQxNGEyMTAwOGViNmRiMTZiYzI2ZTU5OTEzYTM0NmZlMGEwYzczNzgwMzRjNzNhYjIwNjU4NWUzMWUxNjY0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.g-smJ-Bx10s96nheRwLLtvms9oeQXwF5hnfGcItQiK0"> <img alt="截图于 2024-02-07 下午 7:42:06" width="350" src="https://private-user-images.githubusercontent.com/142529543/303447473-b80e6b39-740d-4375-8524-ae596886ac28.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTcwMzkxMjksIm5iZiI6MTc1NzAzODgyOSwicGF0aCI6Ii8xNDI1Mjk1NDMvMzAzNDQ3NDczLWI4MGU2YjM5LTc0MGQtNDM3NS04NTI0LWFlNTk2ODg2YWMyOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkwNVQwMjIwMjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iYmJkYjljOWVhYjhmMzVhMTY3ODc0OWVhNWYzOTg2MDliNTEwN2JmMGVkNmVjMzYzNDRhODkwMWVkOTEyZTYyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.CzEd0_g0vR58xvZGkFd-s71KV6srgugCzMcEx8-1keU"> ![matched_and_unmatched_pair_dist](https://github.com/joonson/syncnet_trainer/assets/142529543/d9ea466b-d6d3-454a-b330-065c33b79b9d)

您好，请问这个问题您最后解决了吗
